pkg/controller/controller.go

```diff
@@ -247,6 +252,12 @@ func (c *controller) runController(params ControllerParams) {
 
 		interval := params.RunInterval
 
+		if params.MinTriggerInterval > 0 {
+			if delayNeeded, delay := c.needsDelay(params.MinTriggerInterval); delayNeeded {
+				time.Sleep(delay)
+			}
+		}
```

Comment:
The child function already checks the zero condition, no need to do it here:
```suggestion
		if delayNeeded, delay := c.needsDelay(params.MinTriggerInterval); delayNeeded {
			time.Sleep(delay)
		}
```